californium-core/src/main/java/org/eclipse/californium/core/CoapServer.java

```diff
@@ -157,13 +157,17 @@ public CoapServer(final NetworkConfig config, final int... ports) {
 		
 		// endpoints
 		this.endpoints = new ArrayList<>();
-		// sets the central thread pool for the protocol stage over all endpoints
-		this.executor = Executors.newScheduledThreadPool(//
-				config.getInt(NetworkConfig.Keys.PROTOCOL_STAGE_THREAD_COUNT), //
-				new Utils.NamedThreadFactory("CoapServer#")); //$NON-NLS-1$
-		// create endpoint for each port
-		for (int port : ports) {
-			addEndpoint(new CoapEndpoint(port, this.config));
+		if(null != config) {
```

Comment:
The solution to this problem is not checking for `null` but using `this.config` instead (it has been initialized to a non-null value in the code above).